components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/PaginatableFilterableApplicationDAO.java

```diff
@@ -46,9 +46,23 @@ ApplicationBasicInfo[] getAllPaginatedApplicationBasicInfo(int pageNumber) throw
      * @param limit  Count value.
      * @return An array of {@link ApplicationBasicInfo} instances within the limit.
      * @throws IdentityApplicationManagementException Error in retrieving basic application information.
+     * @Deprecated This is being deprecated as introducing excludeSystemPortals to exclude system portals in response.
+     * use {@link PaginatableFilterableApplicationDAO#getApplicationBasicInfo(int, int, Boolean)}.
      */
     ApplicationBasicInfo[] getApplicationBasicInfo(int offset, int limit) throws IdentityApplicationManagementException;
 
+    /**
+     * Get all the basic application information based on the offset and the limit.
+     *
+     * @param offset                Starting index of the count.
+     * @param limit                 Count value.
+     * @param excludeSystemPortals  Exclude system portals.
+     * @return An array of {@link ApplicationBasicInfo} instances within the limit.
+     * @throws IdentityApplicationManagementException Error in retrieving basic application information.
+     */
+    ApplicationBasicInfo[] getApplicationBasicInfo(int offset, int limit, Boolean excludeSystemPortals)
```

Comment:
Should be a default method. Check for other places as well.

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/6026/commits/9ab5dadbb9c0b0f8ceffa6a997e49f282954572b